internal/app/configuration/api.go

```diff
@@ -42,12 +43,13 @@ func adminHandler(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
-		proxyConfig := ProxyConfig{}
+		proxyConfig := pactproxy.Config{}
 		err = json.Unmarshal(configBytes, &proxyConfig)
 		if err != nil {
 			httpresponse.Errorf(w, http.StatusBadRequest, "unable to parse interactionConstraint from data. %s", err.Error())
 			return
 		}
+		proxyConfig.SetDefaults()
```

Comment:
Little bit confused about this. We first unmarshal `proxyConfig` from the payload, we then invoke `SetDefaults` which creates a new Struct, and runs `	envconfig.ProcessWith(context.Background(), &defaultConfig, envconfig.MapLookuper(nil))` . does this mean we ignore the unmarshalled result?

Reply:
This is a bit horrible.  I think this admin handler is only used by the tests.  Because the config is posted as json, the `env` loading can't be used.  Any config values not passed will be set to zero values.

`proxyConfig.SetDefaults()` is updating the struct and overwriting zero values with the default values, for fields where this makes sense (AdminPort, WaitDelay, WaitDuration).  Tests fail if WaitDelay is zero.

Reply:
I have removed the SetDefaults as it is indeed confusing.  Moved back to a simpler handling of defaults in the proxy: define constants and a simple `if` to set them if passed values are zero.